### ISSUE TITLE
Day2

### ISSUE THREAD

**Issue by sanshengshi626** [sanshengshi626/daka_advance-machine-learning#2]
GBDT算法梳理

1. 前向分布算法
由于学习的是加法模型，如果能从前向后每一步只学习一个基函数及其系数，逐步逼近优化目标
2. 负梯度拟合
拟合一颗CART回归树，得到了第t颗回归树，其对应的叶节点区域𝑅𝑡𝑗,𝑗=1,2,...,𝐽。其中J为叶子节点的个数。

　　　　针对每一个叶子节点里的样本，我们求出使损失函数最小，也就是拟合叶子节点最好的的输出值𝑐𝑡𝑗如下：
𝑐𝑡𝑗=𝑎𝑟𝑔𝑚𝑖𝑛⏟𝑐∑𝑥𝑖∈𝑅𝑡𝑗𝐿(𝑦𝑖,𝑓𝑡−1(𝑥𝑖)+𝑐)
　　　　这样我们就得到了本轮的决策树拟合函数

3. 损失函数
l=1/2(f-y)^2
4. 回归
回归树总体流程类似于分类树，区别在于，回归树的每一个节点都会得一个预测值
5. 二分类，多分类
而分类：sigmoid变换
多酚类：Softmax变换
6. 正则化
第一种是和Adaboost类似的正则化项，即步长(learning rate)
第二种正则化的方式是通过子采样比例
第三种是对于弱学习器即CART回归树进行正则化剪枝
7. 优缺点
优点：

预测精度高
适合低维数据
能处理非线性数据
可以灵活处理各种类型的数据，包括连续值和离散值。
在相对少的调参时间情况下，预测的准备率也可以比较高。这个是相对SVM来说的。
使用一些健壮的损失函数，对异常值的鲁棒性非常强。比如 Huber损失函数和Quantile损失函数。
缺点：

由于弱学习器之间存在依赖关系，难以并行训练数据。不过可以通过自采样的SGBT来达到部分并行。
如果数据维度较高时会加大算法的计算复杂度
8. sklearn参数
n_estimators: 也就是弱学习器的最大迭代次数
learning_rate: 即每个弱学习器的权重缩减系数𝜈
subsample: 即我们在原理篇的正则化章节讲到的子采样
9. 应用场景
GBDT几乎可用于所有回归问题（线性/非线性），相对logistic regression仅能用于线性回归，GBDT的适用面非常广。亦可用于二分类问题（设定阈值，大于阈值为正例，反之为负例
